net/unicoap: Unified and Modular CoAP Stack: Minimal Client (pt 3) (#22266)
RIOT-OS/RIOT: examples/networking/coap/unicoap_client/README.md

# Sample Client Application With `unicoap`

> This a sample application demonstrating how you send CoAP requests using `unicoap`,

Reviewer: ```suggestion
This is a sample application demonstrating how you send CoAP requests using `unicoap`,
```

the unified and modular CoAP suite in RIOT. Visit the
[`unicoap` client tutorial on doc.riot-os.org](http://doc.riot-os.org/group__net__unicoap__client__tutorial.html)
for a detailed code-level instructions.

## Running the Example on RIOT's Native Board

Create a tap interface (to which RIOT will connect):

```sh
sudo ip tuntap add tap0 mode tap user ${USER}
sudo ip link set tap0 up
```

To try this example on your host, run:
```sh
BOARD=native make flash term
```
This will compile and run the application.

> The application will print a network-layer address.

Reviewer: ```suggestion
```

Not really relevant on the client side I'd say?

In a second terminal session, you can run

```sh
python3 server.py"
```

> to send a CoAP request through the tap interface to the CoAP server.

Reviewer: ```suggestion
Now, you can send CoAP requests from RIOT through the tap interface to the CoAP server.
```

Reviewer: Maybe you also want to briefly mention which endpoints the server supports?

# Sample Client Application With `unicoap`

This is a sample application demonstrating how you send CoAP requests using `unicoap`,
the unified and modular CoAP suite in RIOT. Visit the
[`unicoap` client tutorial on doc.riot-os.org](http://doc.riot-os.org/group__net__unicoap__client__tutorial.html)
for a detailed code-level instructions.

## Running the Example on RIOT's Native Board

Create a tap interface (to which RIOT will connect):

```sh
sudo ip tuntap add tap0 mode tap user ${USER}
sudo ip link set tap0 up
```

To try this example on your host, run:
```sh
BOARD=native make flash term
```
This will compile and run the application.

In a second terminal session, you can run

```sh

> python3 server.py"

Reviewer 2: ```suggestion
python3 server.py
```

```

Now, you can send CoAP requests from RIOT through the tap interface to the CoAP server. The Python
script implements the following resources:
- `/time`: Responds with current timestamp
- `/other/separate`: Sends response in separate message instead of piggybacking (if request has been

> transmitted reliably over UDP or DTLS at all)

Reviewer 2: ```suggestion
- `/other/separate`: Sends response in separate message instead of piggybacking
  (if request has beentransmitted reliably over UDP or DTLS at all)
```
